src/main/java/com/google/sps/servlets/NotificationServlet.java

```diff
@@ -0,0 +1,195 @@
+      prepStatement.executeUpdate();
+    } catch (SQLException ex) {
+      Logger logger = Logger.getLogger(NotificationServlet.class.getName());
+      logger.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+  }
+
+  /**
+   * Receives the data for a notification-user relation and inserts it into
+   * UserNotification.
+   */
+  private void insertToUserNotification(Connection connection, int userId, int notificationId) {
+    String query = "INSERT INTO UserNotification(user_id, notification_id) VALUES(?,?)";
+    try {
+      // Prepare the statement to be inserted.
+      PreparedStatement prepStatement = connection.prepareStatement(query);
+      prepStatement.setInt(SqlConstants.USER_NOTIFICATION_INSERT_USERID, userId);
+      prepStatement.setInt(SqlConstants.USER_NOTIFICATION_INSERT_NOTIFICATIONID, notificationId);
+      prepStatement.executeUpdate();
+    } catch (SQLException ex) {
+      Logger logger = Logger.getLogger(NotificationServlet.class.getName());
+      logger.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+  }
+
+  /**
+   * Returns the ID of a just inserted notification.
+   */
+  private int getLastInsertedNotificationId(Connection connection) {
+    String query =  "SELECT id FROM Notification ORDER BY date_time";
+    // Query the information from Notification table.
+    int notificationId = 0;
+    try (PreparedStatement pst = connection.prepareStatement(query);
+         ResultSet resultSet = pst.executeQuery()) {
+      // Select the id of the last notification to be inserted.
+      resultSet.last();
+      notificationId = resultSet.getInt(1);
+    } catch (SQLException ex) {
+      Logger logger = Logger.getLogger(NotificationServlet.class.getName());
+      logger.log(Level.SEVERE, ex.getMessage(), ex);
+    }
+    return notificationId;
+  }
+
+  /**
+   * Inserts new row to Notification table and adds row to UserNotification of the author's and 
+   * following users' IDs relationship with the last inserted Notification.
+   */
+  private void createNotification(String query, String notificationUrl, String notificationMessage,
+                                  Timestamp localTimestamp) {
+    // Set up connection for insertions and query IDs of users to notify with same connection.
+    try (Connection connection = DriverManager.getConnection(Utility.SQL_LOCAL_URL, 
+            Utility.SQL_LOCAL_USER, Utility.SQL_LOCAL_PASSWORD);
+          PreparedStatement pst = connection.prepareStatement(query);
+          ResultSet resultSet = pst.executeQuery()) {
+      // Insert notification and get its ID to relate in UserNotification table.
+      insertToNotification(connection, notificationMessage, notificationUrl, localTimestamp);
+      int notificationId = getLastInsertedNotificationId(connection);
+      // Iterate through the query's result set to insert all notifications.
+      while (resultSet.next()) {
+        insertToUserNotification(connection, resultSet.getInt(1), notificationId);
```

Comment:
nit: can you make this '1' into a constant?

Reply:
Done!